File: csharp/HyperSwitchAI/src/GetToken.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using DotNetEnv;

namespace HyperSwitchAI
{
    public class AuthRequest
    {
        public string? username { get; set; }
        public string? password { get; set; }
    }

    public class AuthResponse
    {
        public string? token { get; set; }
    }

    public class GetToken
    {
        private const string BASE_URL = "https://api.hyperswitchai.com";

        public static async Task RunAsync()
        {
            try
            {
                // Load environment variables
                Env.Load();

                var username = Environment.GetEnvironmentVariable("USERNAME");
                var password = Environment.GetEnvironmentVariable("PASSWORD");

                if (string.IsNullOrEmpty(username) || string.IsNullOrEmpty(password))
                {
                    throw new Exception("USERNAME and PASSWORD must be set in a .env file in the csharp directory. You can rename .env.sample to .env and fill in your own email and password.");
                }

                using (var client = new HttpClient())
                {
                    var authRequest = new AuthRequest
                    {
                        username = username,
                        password = password
                    };

                    var json = JsonSerializer.Serialize(authRequest);
                    var content = new StringContent(json, Encoding.UTF8, "application/json");

                    var response = await client.PostAsync($"{BASE_URL}/auth", content);

                    if (!response.IsSuccessStatusCode)
                    {
                        throw new Exception("Authentication failed");
                    }

                    var responseBody = await response.Content.ReadAsStringAsync();
                    var authResponse = JsonSerializer.Deserialize<AuthResponse>(responseBody);

                    Console.WriteLine("\u001b[32m✅ Token successfully retrieved\u001b[0m");
                    Console.WriteLine($"Token: {authResponse?.token}");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: java/src/main/java/com/hyperswitchai/examples/GetToken.java
package com.hyperswitchai.examples;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.github.cdimascio.dotenv.Dotenv;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.HashMap;
import java.util.Map;

public class GetToken {
    private static final String BASE_URL = "https://api.hyperswitchai.com";
    private static final ObjectMapper mapper = new ObjectMapper();
    private static final HttpClient client = HttpClient.newHttpClient();
    private static final Dotenv dotenv = Dotenv.load();

    public static void main(String[] args) {
        try {
            // Check for required environment variables
            String username = dotenv.get("USERNAME");
            String password = dotenv.get("PASSWORD");
            
            if (username == null || password == null) {
                throw new RuntimeException("USERNAME and PASSWORD must be set in a .env file in the java directory. " +
                    "You can rename .env.sample to .env and fill in your own email and password.");
            }

            // Prepare authentication request
            Map<String, String> authPayload = new HashMap<>();
            authPayload.put("username", username);
            authPayload.put("password", password);

            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + "/auth"))
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(mapper.writeValueAsString(authPayload)))
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            if (response.statusCode() != 200) {
                throw new RuntimeException("Authentication failed: " + response.body());
            }

            Map<String, String> authResponse = mapper.readValue(response.body(), Map.class);
            String token = authResponse.get("token");

            System.out.println("\u001B[32m✅ Token successfully retrieved\u001B[0m");
            System.out.println("Token: " + token);

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
} 